### PR TITLE
Feature/flip volume

### PR DESCRIPTION
**Pull request by toloudis**
Estimated time to review:  ~15 min ?  A lot of code changes but all small.

Allows users to flip (mirror / invert) each of the three x,y,z axes for the volume.

AGAVE 1 6 0 2024-05-02 11-27-13

The internal storage uses -1 for a flipped axis, and 1 for an unflipped one.

The volume sampling shader has to switch its sampling wrap mode to "REPEAT" to make this work.

When adding a new feature like this, we need to make sure the feature is implemented in json save files (serialization), python script output, as well as client/server communication (aka Commands). 

In order to make it easier to add new serialized variables, without having to bump versions, I had to upgrade the json library I am using.   Disregard changes to json.hpp as it is a third party file.

### CODE REVIEW

File: renderlib/json/json.hpp
Comment (toloudis) [inline renderlib/json/json.hpp:1]: can disregard diff in this file as it is a third party include which I updated

File: agave_app/Serialize.h [hunk 1]
```diff
@@ -106,28 +107,29 @@ struct ViewerState
...
-                                 camera,
-                                 backgroundColor,
-                                 boundingBoxColor,
-                                 showBoundingBox,
-                                 channels,
-                                 density,
-                                 lights,
-                                 capture,
-                                 showScaleBar)
+  NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT(ViewerState,
```
Comment (toloudis) [inline agave_app/Serialize.h:115]: comment why this macro is here and why it usees WITH_DEFAULT

File: renderlib/ImageXYZC.cpp [hunk 2]
```diff
@@ -79,9 +80,21 @@ ImageXYZC::maxPixelDimension() const
...
-  m_scaleX = x;
-  m_scaleY = y;
-  m_scaleZ = z;
+  m_scaleX = abs(x);
+  m_scaleY = abs(y);
+  m_scaleZ = abs(z);
+}
+
+void
+ImageXYZC::setVolumeAxesFlipped(float x, float y, float z)
```
Comment (toloudis) [inline renderlib/ImageXYZC.cpp:89]: this really should be storing ints or booleans

File: agave_app/AppearanceSettingsWidget.cpp [hunk 3]
```diff
@@ -369,6 +384,34 @@ QAppearanceSettingsWidget::createSkyLightingControls()
   return section;
 }
 
+void
+QAppearanceSettingsWidget::OnFlipX(bool value)
```
Comment (meganrm) [inline agave_app/AppearanceSettingsWidget.cpp:388]: just making a note of my suggestion to change these 3 functions to one that takes in an `axis` value to reduce the copied code

**Review by meganrm — APPROVED** [review]
Some small suggestions came up in our review meeting, but overall seems good

**Review by rugeli — APPROVED** [review]
Looks great!

**Pull request merged**